packages/framework/src/Framework/Features/Publications/PublicationService.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hyde\Framework\Features\Publications;
+
+use function basename;
+use function dirname;
+use Exception;
+use function glob;
+use Hyde\Framework\Features\Publications\Models\PublicationType;
+use Hyde\Hyde;
+use Hyde\Pages\PublicationPage;
+use Illuminate\Support\Str;
+use Rgasch\Collection\Collection;
+use function Safe\file_get_contents;
+use Spatie\YamlFrontMatter\YamlFrontMatter;
+
+/**
+ * @see \Hyde\Framework\Testing\Feature\Services\PublicationServiceTest
+ */
+class PublicationService
+{
+    /**
+     * Return a collection of all defined publication types, indexed by the directory name.
+     *
+     * @todo We might want to refactor to cache this in the Kernel, maybe under $publications?
+     *
+     * @return Collection<string, PublicationType>
+     */
+    public static function getPublicationTypes(): Collection
+    {
+        return Collection::create(static::getSchemaFiles())->mapWithKeys(function (string $schemaFile): array {
+            $publicationType = PublicationType::fromFile(Hyde::pathToRelative($schemaFile));
+
+            return [$publicationType->getDirectory() => $publicationType];
+        });
+    }
+
+    /**
+     * Return all publications for a given publication type.
+     */
+    public static function getPublicationsForPubType(PublicationType $pubType): Collection
+    {
+        return Collection::create(static::getPublicationFiles($pubType->getDirectory()))->map(function (string $file): PublicationPage {
+            return static::parsePublicationFile(Hyde::pathToRelative($file));
+        });
+    }
+
+    /**
+     * Return all media items for a given publication type.
+     */
+    public static function getMediaForPubType(PublicationType $pubType): Collection
+    {
+        return Collection::create(static::getMediaFiles($pubType->getDirectory()))->map(function (string $file): string {
+            return Hyde::pathToRelative($file);
+        });
+    }
+
+    /**
+     * Parse a publication Markdown source file and return a PublicationPage object.
+     *
+     * @param  string  $identifier  Example: my-publication/hello.md or my-publication/hello
+     */
+    public static function parsePublicationFile(string $identifier): PublicationPage
+    {
+        $identifier = Str::replaceLast('.md', '', $identifier);
+        $fileData = static::getFileData("$identifier.md");
+
+        $parsedFileData = YamlFrontMatter::markdownCompatibleParse($fileData);
+
+        return new PublicationPage(
+            type:       PublicationType::get(dirname($identifier)),
+            identifier: basename($identifier),
+            matter:     $parsedFileData->matter(),
+            markdown:   $parsedFileData->body()
+        );
+    }
```

Comment:
This logic should be handled in the SourceFileParser class

Reply:
Resolved in https://github.com/hydephp/develop/commit/55615d7b14a7464e6ffc19d14100f7e3f4916c7d